c2rust-analyze/src/borrowck/type_check.rs

```diff
@@ -299,6 +299,15 @@ impl<'tcx> TypeChecker<'tcx, '_> {
 
             Rvalue::UnaryOp(_, ref op) => self.visit_operand(op),
 
+            Rvalue::Repeat(ref op, _) => {
+                if op.ty(self.local_decls, self.tcx).is_any_ptr() {
```

Comment:
I think we have to be careful about what we consider pointers throughout `c2rust-analyze`.

- `Ty::is_any_ptr` checks `Ref`, `RawPtr`, and `FnPtr`
- `assign_pointer_ids` checks `Ref` and `RawPtr` manually
- and `is_trivial` checks just `RawPtr`

Reply:
I think that's fair, but here we simply just don't want any of `Ref`, `RawPtr`, or `FnPtr`.

Reply:
This should be fine for this PR, but I think it's something we should discuss at some point to clarify each of the definitions.